agent/auto-config/run.go

```diff
@@ -122,8 +122,11 @@ func (ac *AutoConfig) run(ctx context.Context, exit chan struct{}) {
 	// this point, auto_encrypt was just being setup
 	// successfully.
 	calcFallbackInterval := func() time.Duration {
-		certExpiry := ac.acConfig.TLSConfigurator.AutoEncryptCertNotAfter()
-		return certExpiry.Add(ac.acConfig.FallbackLeeway).Sub(time.Now())
+		cert := ac.acConfig.TLSConfigurator.AutoEncryptCert()
+		if cert == nil {
+			return -1
+		}
+		return cert.NotAfter.Add(ac.acConfig.FallbackLeeway).Sub(time.Now())
```

Comment:
I think this line got harder to parse now, what if there was an intermediate variable here?

```suggestion
		renewalDeadline := cert.NotAfter.Add(ac.acConfig.FallbackLeeway)
		return renewalDeadline.Sub(time.Now())
```